### PR TITLE
Throw UnsupportedOpenAPISpec when openAPI version is not 3.x.x

### PR DESCRIPTION
**Pull request by damey2011**
The current implementation does not specify the supported openAPI spec versions. This MR:

- Documents that it supports version 3.x.x
- Throws an error when openAPI spec of non 3.x.x is provided

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ Please check the [issue tracker](https://github.com/sennder/python-client-genera
 or feature request before creating a new issue.
 
 ### Setting up your local environment
+This project was developed using Python3.8, other python version have not been fully tested.
```
Comment (bogdanp05) [inline .github/CONTRIBUTING.md:10]: Can you replace this with "older Python versions..."? In the pipeline we do have tests for 3.9 and 3.10.

File: README.md [hunk 2]
```diff
@@ -14,6 +14,7 @@ flowchart LR
     generator -- "generates" --> package
 ```
 
+> :warning: **Currently does not support OpenAPI 2.0**: MR for < 3.0 support are welcome
```
Comment (bogdanp05) [inline README.md:17]: MRs*

**Review by bogdanp05 — CHANGES_REQUESTED** [review]
Thanks for the MR! Looks good, just 2 minor things.

I don't fully get why the pipeline didn't run, probably because it's coming from a fork? Anyways, you'll have to change the last commit message to smth like "test: ..." so it passes commitlint.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (damey2011) [reply .github/CONTRIBUTING.md:10]: Okay.. But for some reasons `poetry install` fails while installing `typed-ast` with python3.9 and 3.10 for me.

```
  PoetryException

  Failed to install /Users/theoluwanifemi/Library/Caches/pypoetry/artifacts/cc/50/4d/e1b50dadd10f8d3f8bd6f6c2e764cf988f0e86c4047c0210353ba77680/typed_ast-1.4.3.tar.gz

  at ~/Library/Application Support/pypoetry/venv/lib/python3.10/site-packages/poetry/utils/pip.py:58 in pip_install
       54│ 
       55│     try:
       56│         return environment.run_pip(*args)
       57│     except EnvCommandError as e:
    →  58│         raise PoetryException(f"Failed to install {path.as_posix()}") from e
       59│ 


```

File: README.md (hunk 2 above)
Comment (damey2011) [reply README.md]: Fixed.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (bogdanp05) [reply .github/CONTRIBUTING.md:10]: That's weird. I tried just now with python 3.10.9 and it worked. And it does work in the [pipeline](https://github.com/sennder/python-client-generator/blob/main/.github/workflows/cicd.yml#L23).

**Pull request merged**